### PR TITLE
SSLStream - Fix write renegotiation and make a single function for it

### PR DESCRIPTION
**Pull request by Drawaes**
I was taking a hard look at the code for triggering the callback after an async renegotiation for both read and write. The write side is currently missing the TaskCompletionSource call that was added, but seemed to drop out of my PR. 

I have put this in and moved the duplicate code to a single function which is what I was prototyping on the read side to make that full async.

### CODE REVIEW

**Comment by Drawaes** [comment]
As an aside I am wracking my brain on how we can test this code flow better, but the only way with .net currently might just be some sort of unit tests that artificially get in there and force the situation.

Hoping someone can come up with ideas on this?

**Comment by Drawaes** [comment]
@dotnet-bot Test Outerloop Windows x64 Debug Build
@dotnet-bot Test Outerloop UWP CoreCLR x64 Debug Build
@dotnet-bot Test Outerloop Linux x64 Release Build

**Comment by stephentoub** [comment]
cc: @geoffkizer

File: src/System.Net.Security/src/System/Net/Security/SslState.cs [hunk 1]
```diff
@@ -1346,25 +1346,27 @@ internal void FinishWrite()
...
+            object obj = _queuedWriteStateRequest;
+            _queuedWriteStateRequest = null;
+            switch (obj)
+            {
+                case null:
+                    break;
+                case LazyAsyncResult lazy:
+                    lazy.InvokeCallback();
+                    break;
+                case TaskCompletionSource<int> tsc:
```
Comment (stephentoub) [inline src/System.Net.Security/src/System/Net/Security/SslState.cs]: Nit: tcs?

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Linux x64 Release Build please

**Pull request merged**